radixdlt/src/main/java/com/radixdlt/consensus/bft/VertexStore.java

```diff
@@ -521,7 +521,8 @@ public VertexMetadata insertVertex(Vertex vertex) throws VertexInsertionExceptio
 		}
 
 		for (Vertex committed : path) {
-			CommittedAtom committedAtom = new CommittedAtom(committed.getAtom(), commitMetadata);
+			long timestamp = committed.getQC().getTimestampedSignatures().weightedTimestamp();
```

Comment:
timestamp really needs to be inside vertexmetadata as it needs to be part of the three rounds of voting. In this case getQC().getTimestampedSignatures().weightedTimestamp() might be different for all nodes.